### PR TITLE
Added example dataset installation as resource

### PR DESCRIPTION
**Pull request by Skarlso**

### CODE REVIEW

File: pkg/provider.go [hunk 1]
```diff
@@ -68,11 +68,12 @@ func Provider() *schema.Provider {
...
-			"oasis_ipwhitelist":   resourceIPWhitelist(),
-			"oasis_certificate":   resourceCertificate(),
-			"oasis_backup_policy": resourceBackupPolicy(),
-			"oasis_project":       resourceProject(),
+			"oasis_deployment":      resourceDeployment(),
+			"oasis_ipwhitelist":     resourceIPWhitelist(),
+			"oasis_certificate":     resourceCertificate(),
+			"oasis_backup_policy":   resourceBackupPolicy(),
+			"oasis_project":         resourceProject(),
+			"oasis_example_dataset": resourceExampleDataset(),
```
Comment (ewoutp) [inline pkg/provider.go:76]: Shouldn't this be `oasis_example_dataset_installation` ?

Comment (Skarlso) [reply pkg/provider.go:76]: I think it should, yes.

File: examples/creating_example_dataset_installation.tf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * will be loaded in a random generated db name ( which will be displayed in the status output ).
+ */
+provider "oasis" {
+  api_key_id     = ""
+  api_key_secret = ""
+}
+
+// Use an existing project
+data "oasis_project" "my_project" {
+  id = ""
```
Comment (ewoutp) [inline examples/creating_example_dataset_installation.tf:14]: Add comment to indicat that they need to enter their project id here

File: examples/creating_example_dataset_installation.tf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * Create a deployment.
+ */
+resource "oasis_deployment" "my_oneshard_deployment" {
+  name    = "Test Terraform Deployment"
+  project = data.oasis_project.my_project.id
+  location {
+    region = "gcp-europe-west4"
+  }
+  version {
+    db_version = "3.6.0"
```
Comment (ewoutp) [inline examples/creating_example_dataset_installation.tf:27]: 3.6.2.2

**Pull request merged**